src/database/schema/ChannelListSchema.ts

```diff
@@ -216,12 +219,55 @@ class ChannelList implements BaseDbSchema {
     return results.rows.raw().map(ChannelList.fromDatabaseObject);
   }
 
-  static async getUnreadCount(db: SQLiteDatabase): Promise<number> {
+  static async getAllAnonymousChannelList(
+    db: SQLiteDatabase,
+    myId: string,
+    myAnonymousId: string
+  ): Promise<ChannelList[]> {
+    const [results] = await db.executeSql(
+      `SELECT 
+        A.*,
+        CASE last_updated_by
+          WHEN ? THEN 1
+          WHEN ? THEN 1
+          ELSE 0 END AS is_me,
+        B.id AS user_row_id,
+        B.user_id,
+        B.channel_id,
+        B.username,
+        B.country_code,
+        B.created_at AS user_created_at,
+        B.updated_at AS user_updated_at,
+        B.last_active_at,
+        B.profile_picture,
+        B.bio,
+        B.is_banned
+      FROM ${ChannelList.getTableName()} A
+      LEFT JOIN ${UserSchema.getTableName()} B
+      ON A.last_updated_by = B.user_id AND A.id = B.channel_id
+      WHERE (expired_at IS NULL OR datetime(expired_at) >= datetime('now')) AND A.description != ''
+      AND A.channel_type IN ('ANON_PM', 'ANON_POST_NOTIFICATION', 'ANON_GROUP')
+      ORDER BY last_updated_at DESC`,
+      [myId, myAnonymousId]
+    );
+    return results.rows.raw().map(ChannelList.fromDatabaseObject);
+  }
+
+  static async getUnreadCount(
+    db: SQLiteDatabase,
+    channelCategory: 'SIGNED' | 'ANON'
+  ): Promise<number> {
     try {
-      const [results] = await db.executeSql(
-        `SELECT SUM(unread_count) as unread_count FROM ${ChannelList.getTableName()}
-        WHERE expired_at IS NULL OR datetime(expired_at) >= datetime('now')`
-      );
+      const signedQuery = "('PM', 'GROUP', 'TOPIC', 'POST_NOTIFICATION', 'FOLLOW')";
+      const anonQuery = "('ANON_PM', 'ANON_GROUP', 'ANON_POST_NOTIFICATION')";
+
+      const selectQuery = `SELECT SUM(unread_count) as unread_count
+        FROM ${ChannelList.getTableName()}
+        WHERE channel_type IN ${channelCategory === 'ANON' ? anonQuery : signedQuery}
+        AND expired_at IS NULL
+        OR datetime(expired_at) >= datetime('now')`;
+
+      const [results] = await db.executeSql(selectQuery);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This new method `getAllAnonymousChannelList` seems to be well implemented with proper SQL query and parameters. However, there's a potential SQL injection vulnerability in lines 261 and 262 where the `signedQuery` and `anonQuery` are directly concatenated into the SQL query string in line 266. It would be better to use parameterized queries or prepared statements to avoid this issue.



<!-- This is an auto-generated comment by OSS CodeRabbit -->